src/mon/MonitorDBStore.h

```diff
@@ -709,7 +709,7 @@ class MonitorDBStore
   }
 
   void close() {
-    // there should be no work queued!
+    io_work.wait_for_empty();
```

Comment:
Considering the comment says `there should be no work queued!` quite explicitly, I would expect a new comment explaining why we are now waiting for it to empty.

Reply:
After increasing finisher's debug level and found out that the queue was indeed empty when closing.
```
2023-07-20T09:18:50.796+0000 7fcf5c740cc0 10 finisher(0x5570504d85a0) wait_for_empty empty
2023-07-20T09:18:50.796+0000 7fcf5c740cc0 10 finisher(0x5570504d85a0) stop
2023-07-20T09:18:50.796+0000 7fcf467ef700 10 finisher(0x5570504d85a0) finisher_thread stop
2023-07-20T09:18:50.796+0000 7fcf5c740cc0 10 finisher(0x5570504d85a0) stop finish
2023-07-20T09:18:50.796+0000 7fcf5c740cc0  4 rocksdb: [db/db_impl/db_impl.cc:496] Shutdown: canceling all background work
2023-07-20T09:18:50.796+0000 7fcf5c740cc0  4 rocksdb: [db/db_impl/db_impl.cc:704] Shutdown complete
2023-07-20T09:18:50.796+0000 7fcf5c740cc0  0 ceph-mon: created monfs at /var/lib/ceph/mon/ceph-a for mon.a
```
I added an assertion instead (38b94fba937e54a9ccdc435e1903ee785a50a9ab).